### DISCUSSION TITLE
OneOf doesn't produce methods for deserializing

### DISCUSSION THREAD

**Discussion by EmilioMirasola** [oapi-codegen/oapi-codegen#965] [Q&A]
Hi,

I have a problem where oneOf does not produce the helper methods I'm expecting.
My request body definition is as follows:
![image](https://user-images.githubusercontent.com/79855253/218771199-a028bd03-b150-4698-b110-c73953013b6c.png)

My components are as follows:
![image](https://user-images.githubusercontent.com/79855253/218770266-28846e00-2f19-4103-b478-588da91c4727.png)

However, this results in the following code:
![image](https://user-images.githubusercontent.com/79855253/218770474-97f1f8a5-8308-4697-9bf0-28d16ef9699d.png)

Where there aren't any methods on any of the types.

The guide states that given OneOf on components Dog/Cat, methods for parsing body as either are generated.

Do I have an issue with my YAML, my codegen configuration, or is there a library issue?

**Comment by mminklet** [comment]
We are also coming up against this problem. Seems there is a bug in the RESPONSE logic that is being addressed, I kind of assume that it's the same for the request logic too
https://github.com/oapi-codegen/oapi-codegen/issues/970
